### PR TITLE
fix(remix-dev/vite, server-runtime): pass Vite server errors to `vite.ssrFixStacktrace`

### PR DESCRIPTION
**Pull request by hi-ogawa**
Closes: https://github.com/remix-run/remix/issues/8065

I added `DevServerHooks.fixStacktrace` to pass `ViteDevServer.ssrFixStacktrace` to server runtime, then server runtime uses it to extend default behavior of `handleError` during `vite dev`.

## Testing Strategy

I verified the local build on the reproduction repo https://github.com/hi-ogawa/remix-vite-stacktrace-repro/pull/1
I also added playwright test to check error stacktrace line/column in dev error boundary.

### CODE REVIEW

File: packages/remix-dev/vite/plugin.ts [hunk 1]
```diff
@@ -682,7 +682,13 @@ export const remixVitePlugin: RemixVitePlugin = (options = {}) => {
               url
             );
           },
-          fixStacktrace: vite.ssrFixStacktrace,
+          // If an error is caught within the request handler, let Vite fix the
+          // stack trace so it maps back to the actual source code
+          processRequestError: (error) => {
```
Comment (markdalgleish) [inline packages/remix-dev/vite/plugin.ts:687]: Heads up that I've renamed this hook from `fixStacktrace` to `processRequestError`. We want to make sure that all dev server hooks are named in a Vite-agnostic way. As part of this decoupling from Vite, I've also made all of the hooks optional.

File: packages/remix-server-runtime/server.ts [hunk 2]
```diff
@@ -95,10 +95,10 @@ export const createRequestHandler: CreateRequestHandlerFunction = (
 
     let matches = matchServerRoutes(routes, url.pathname);
     let handleError = (error: unknown) => {
-      let devServerHooks = getDevServerHooks();
-      if (devServerHooks && error instanceof Error) {
-        devServerHooks.fixStacktrace(error);
+      if (mode === ServerMode.Development) {
```
Comment (markdalgleish) [inline packages/remix-server-runtime/server.ts:98]: We want to make sure these hooks are never called outside of dev, especially since the hooks are coming from a global that can be modified by anyone.

File: integration/vite-dev-test.ts [hunk 3]
```diff
@@ -385,6 +418,29 @@ test.describe("Vite dev", () => {
...
+    page.on("pageerror", (error) => pageErrors.push(error));
+
+    await page.goto(
+      `http://localhost:${devPort}/error-stacktrace?crash-server-render`
+    );
+    await expect(page.locator("main")).toContainText(
+      "Error: crash-server-render"
+    );
+    await expect(page.locator("main")).toContainText(
+      "error-stacktrace.tsx:16:11"
```
Comment (markdalgleish) [inline integration/vite-dev-test.ts:432]: This is really nice 👍

**Review by markdalgleish — APPROVED** [review]
Nice one! Thanks for the PR 👍

File: packages/remix-dev/vite/plugin.ts (hunk 1 above)
Comment (hi-ogawa) [reply packages/remix-dev/vite/plugin.ts:687]: I don't usually have a strong opinion (and good sense) on naming things, so please feel free and don't hesitate to tweak anything I named.

> decoupling from Vite

That sounds like a huge plan!
I'll keep in mind that when doing things in the future.

Comment (markdalgleish) [reply packages/remix-dev/vite/plugin.ts:687]: We don't see any need for these specific hooks to be used outside of Vite. For now it's mainly just to avoid the weirdness of having references to Vite in `@remix-run/server-runtime` 😅

Comment (lilouartz) [reply packages/remix-dev/vite/plugin.ts:687]: I am getting errors such as:

```
Error: `line` must be greater than 0 (lines start at line 1)
```

and I am wondering if related to these changes.
